src/server_requests.js

```diff
@@ -111,6 +111,44 @@ export function fetchDatastackFromFile(payload) {
   );
 }
 
+/**
+ * Get a list of the column names of a vector file.
+ *
+ * @param {string} payload - path to file
+ * @returns {Promise} resolves array
+ */
+export async function getVectorColumnNames(payload) {
+  const result = await (
+    fetch(`${HOSTNAME}:${process.env.PORT}/colnames`, {
+      method: 'post',
+      body: JSON.stringify({vector_path: payload}),
+      headers: { 'Content-Type': 'application/json' },
+    })
+      .then((response) => response.json())
+      .catch((error) => logger.error(error.stack))
+  );
+  return result.colnames;
+}
+
+/**
+ * Get a boolean indicating whether a vector file may contain points.
+ *
+ * @param {string} payload - path to file
+ * @returns {Promise} resolves boolean
+ */
+export async function getVectorHasPoints(payload) {
+  const result = await (

```

Comment:
Same reasoning as above, I wanted it to return just the boolean, so that other functions outside of `server_requests.js` don't have to know anything about the server API. But now I realize that accessing the `has_points` property can go in a `.then` clause. I'll try making these return Promises to be consistent with the others.